### DISCUSSION TITLE
Creating a DirectMARLEnv from a single URDF

### DISCUSSION THREAD

**Discussion by jackzeng-robotics** [isaac-sim/IsaacLab#1498] [Q&A]
Hi all, 

I have been developing some envs for a multi-agent setup, but with a single URDF. So essentially, is it treated as a single agent with higher observation and action spaces. I want to use the true multi-agent algorithms such as MAPPO, so I want to rewrite it as a DirectMARL environment. My problem is that the setup is constrained, because it looks like this:

![image](https://github.com/user-attachments/assets/102cd13c-a028-49dc-ae0c-191757ed0f16)

It's 3 drones carrying an object together. The attachments for the ropes are modeled in the URDF so I would also have to remodel that, and I'm not sure how to do that with the InteractiveScene. Is there a way to either keep the model and treat each drone as a separate Articulation in some way? Or is there a good way to make 3 separate Articulations, a basket object and link them together using cables in the InteractiveScene?

Thanks in advance.

Kind regards,
Jack

**Comment by RandomOakForest** [comment]
Thanks for posting this. The team is picking up on this example and we hope to have a posted answer soon.

**Reply by jackzeng-robotics** [reply]
Thanks for letting me know, looking forward to it!
